komodo/show_version.py

```diff
@@ -0,0 +1,180 @@
+        config.read_string("[MAIN]\n" + stream.read())
+    return dict(config["MAIN"])
+
+
+def get_komodoenv_path(release: str) -> Path:
+    """
+    Use the release name to find the 'real' release path, but from a
+    komodoenv environment. These environments overwrite the environment
+    variable $KOMODO_RELEASE and $PATH, but they store the original path
+    in the komodoenv at 'root/pyvenv.cfg'.
+
+    Args:
+        release: The name of the release, e.g. as stored in the KOMODO_RELEASE
+            environment variable.
+
+    Returns:
+        The path to the release, where the 'root/bin' folder is.
+    """
+    config = read_config(Path(release) / "root" / "pyvenv.cfg")
+    path, *_ = config["home"].split("/root/bin")
+    return Path(path)
+
+
+def get_komodo_path(release: str) -> Path:
+    """
+    Use the release name to find the 'real' release path in an ordinary
+    komodo environment. E.g., the release may be something like
+    '2023.01.02-py38' but the 'real' release (where the release manifest
+    is stored) might be platform-specific, e.g. '2023.01.02-py38-rhel7'.
+    The real path is in the PATH, so we try to get it from there.
+
+    Args:
+        release: The name of the release, e.g. from $KOMODO_RELEASE.
+
+    Returns:
+        The path to the release, where the 'root/bin' folder is.
+    """
+    paths = os.environ["PATH"]
+    pattern = re.compile(rf"(.*?{release}.*?)/root/bin")
+    result = pattern.search(paths)
+    if result is not None:
+        (path,) = result.groups()
+    else:
+        message = f"Could not retrieve the path to the release {release}."
+        raise RuntimeError(message)
+    return Path(path)
+
+
+def get_version(pkg: str, manifest: Optional[Dict] = None) -> str:
+    """
+    Get the release number (or git commit hash) for a package in a
+    komodo release file. If no file is specified, the current release
+    is used. If no environment is active, the path to the release
+    must be specified.
+
+    Args:
+        pkg: The name of the package to get the version for.
+        manifest: Optional. A mapping of packages to dicts of version and
+            maintainer. Typically, this will come from the YAML file created
+            by komodo when it builds a release.
+
+    Returns:
+        The version number or git hash of the version.
+    """
+    path = None
+    if manifest is None:
+        release = get_release()
+        if (Path(release) / "komodoenv.conf").is_file():
+            path = get_komodoenv_path(release)
+        else:
+            path = get_komodo_path(release)
+        release_file = path.parts[-1]
+
+        with open(path / release_file, "rt", encoding="utf-8") as stream:
+            manifest = yaml.safe_load(stream)
+
+    package = manifest.get(pkg)
+
+    if package is None:
+        path_str = f" {path / release_file}" if path is not None else ""
+        message = f"The package {pkg} is not found in the manifest file{path_str}."
```

Comment:
Missing a blank space in `file{path_str}`, correct?

Reply:
It does look weird but it's correct, because sometimes the sentence ends there.